### PR TITLE
PICARD-2282: Add maintenance options to backup and restore configuration INI files

### PR DESCRIPTION
**Pull request by rdswift**
# Summary

* This is a…
  * [ ] Bug fix
  * [x] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**: Add new maintenance options to backup and restore the INI file.

# Problem

Users requested the ability to easily backup and restore their configuration files.

* JIRA ticket (_optional_): PICARD-2282

# Solution

Add two new options to the "Options" --> "Advanced" --> "Maintenance" page to allow saving the options INI file to a user-selected backup file name, and to restore the settings from a selected backup file.  When restoring, the backup file overwrites the current configuration file, so the current file is backed up automatically.

# Screen Shots

image

image

# Action

Update documentation to include the new functionality.

### CODE REVIEW

File: picard/config.py [hunk 1]
```diff
@@ -479,3 +491,14 @@ def get_config():
     Config objects for threads are created on demand and cached for later use.
     """
     return config
+
+
+def load_new_config(filename=None):
+    ini_file = get_config().fileName()
```
Comment (zas) [inline picard/config.py:497]: We should use a format-agnostic variable name instead like `config_file`.
Configuration files format may change over the time.

**Review by zas — CHANGES_REQUESTED** [review]

File: picard/ui/options/maintenance.py [hunk 2]
```diff
@@ -154,6 +161,116 @@ def open_config_dir(self):
...
+        else:
+            self._backup_error(dialog_title)
+
+    def load_backup(self):
+        dialog_title = _("Load Backup Configuration File")
+        dialog = QtWidgets.QMessageBox(
+            QtWidgets.QMessageBox.Icon.Warning,
+            dialog_title,
+            _("Loading a backup configuration file will replace the current configuration settings. "
+            "A backup copy of the current file will be saved automatically.\n\nDo you want to continue?"),
```
Comment (rdswift) [inline picard/ui/options/maintenance.py:229]: @zas Is this an acceptable way to combine parts of a long string for translation (so as not to have an extremely long line)?

Comment (zas) [reply picard/ui/options/maintenance.py:229]: Yes, they'll appear as one string in .po files

**Review by zas — APPROVED** [review]
LGTM

**Review by phw — APPROVED** [review]
This works great, thanks a lot for this addition

**Pull request merged**